utils/build-script-impl

```diff
@@ -2113,7 +2113,11 @@ for host in "${ALL_HOSTS[@]}"; do
 
     for product in "${PRODUCTS[@]}"; do
         # Check if we should perform this action.
-        if ! [[ $(should_execute_action "${host}-${product}-build") ]]; then
+        tmp_product=${product}
+        if [ ${tmp_product} == "libdispatch_static" ]; then
```

Comment:
Please don't mix and match sh and bash.  Use either `[[` and `==` or `[` and `=`.

Reply:
Huh, I never realised they were different for `sh` v `bash`. Anyway, fixed.

Reply:
Yeah, POSIX sh only has `[` and `=`.  Bash extended that syntax to have ... something which behaved marginally more expectantly by people at that time.  I would posit that the expectations of people around the behaviour of the conditional checks have changed, so they are just as confusing to many people today.

Reply:
Yeah, POSIX sh only has `[` and `=`.  Bash extended that syntax to have ... something which behaved marginally more expectantly by people at that time.  I would posit that the expectations of people around the behaviour of the conditional checks have changed, so they are just as confusing to most people today.